dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/WorkflowItemRestRepository.java

```diff
@@ -0,0 +1,335 @@
+        return page;
+    }
+
+    @SearchRestMethod(name = "findBySubmitter")
+    public Page<WorkflowItemRest> findBySubmitter(@Param(value = "uuid") UUID submitterID, Pageable pageable) {
+        List<XmlWorkflowItem> witems = null;
+        try {
+            Context context = obtainContext();
+            EPerson ep = epersonService.find(context, submitterID);
+            witems = wis.findBySubmitter(context, ep);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        Page<WorkflowItemRest> page = utils.getPage(witems, pageable).map(converter);
+        return page;
+    }
+
+    @Override
+    protected WorkflowItemRest createAndReturn(Context context) {
+        XmlWorkflowItem source;
+        try {
+            source = submissionService.createWorkflowItem(context, getRequestService().getCurrentRequest());
+        } catch (SQLException | AuthorizeException | WorkflowException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        //if the item go directly in published status we have to manage a status code 204 with no content
+        if (source.getItem().isArchived()) {
+            return null;
+        }
+        return converter.convert(source);
+    }
+
+    @Override
+    protected WorkflowItemRest save(Context context, WorkflowItemRest wfi) {
+        SubmissionConfig submissionConfig =
+            submissionConfigReader.getSubmissionConfigByName(submissionConfigReader.getDefaultSubmissionConfigName());
+        XmlWorkflowItem source = converter.toModel(wfi);
+        for (int stepNum = 0; stepNum < submissionConfig.getNumberOfSteps(); stepNum++) {
+
+            SubmissionStepConfig stepConfig = submissionConfig.getStep(stepNum);
+            /*
+             * First, load the step processing class (using the current
+             * class loader)
+             */
+            ClassLoader loader = this.getClass().getClassLoader();
+            Class stepClass;
+            try {
+                stepClass = loader.loadClass(stepConfig.getProcessingClassName());
+
+                Object stepInstance = stepClass.newInstance();
+
+                if (stepInstance instanceof AbstractProcessingStep) {
+                    // load the JSPStep interface for this step
+                    AbstractProcessingStep stepProcessing = (AbstractProcessingStep) stepClass.newInstance();
+                    stepProcessing.doPreProcessing(context, source);
+                } else {
+                    throw new Exception(
+                        "The submission step class specified by '" + stepConfig.getProcessingClassName() +
+                        "' does not extend the class org.dspace.submit.AbstractProcessingStep!" +
+                        " Therefore it cannot be used by the Configurable Submission as the <processing-class>!");
+                }
+
+            } catch (Exception e) {
+                log.error(e.getMessage(), e);
```

Comment:
This error seems to be caught within a for loop.  Should it be re-thrown instead of being logged?  Can this method proceed if an error occurs here?

Reply:
I removed the method. It was an initial draft of the PUT implementation that was not finalized and not required by the contract